### DISCUSSION TITLE
Provide a custom random number generator for Faker

### DISCUSSION THREAD

**Discussion by xDivisionByZerox** [faker-js/faker#2308] [Polls]
We currently have the issue #2195 which asks for a way to provide a custom random number generator to Faker. Currently Faker uses an implementation of [Mersenne Twister](https://en.wikipedia.org/wiki/Mersenne_Twister). 
We totally see the need to sometimes provide your own implementation over our internal one. Because of that we want to make it possible to do exactly that.

But as all good developers, we do have problems with naming things as well.
Our first instinct was to name it `PRNG`, as it is the technical correct term. But we got concerened that `PRNG` might not be as intuitive to most developers. `RNG` on the other hand side might be more intuitve, but is technically not correct.
We also thought about naming it `random`. The reason being that the provided value is used to provide all the "randomness" in Faker.

If you have any other suggestions, we encourage you to leave a comment and share your insight. Every kind of feedback is welcome as it contributes to the ongoing enhancement of Faker.

**Comment by ST-DDT** [comment]
How about `Random`?

**Comment by Shinigami92** [comment]
`seeder`

**Reply by Shinigami92** [reply]
After having a look into https://github.com/boo1ean/mersenne-twister#usage and some sleep, I think I would also be satisfied with just `generator`

**Comment by softwaredeveloptam** [comment]
RandomGenerator 👍

**Comment by danielroe** [comment]
- `randomizer`

**Reply by Shinigami92** [reply]
@ST-DDT, @danielroe, @JessicaSachs and @Shinigami92 are enough votes and we agree ✅
